src/propagators/text_map_codec.js

```diff
@@ -56,21 +57,28 @@ export default class TextMapCodec {
     }
 
     extract(carrier: any): ?SpanContext {
-        let header = carrier[this._contextKey];
-        let spanContext: ?SpanContext = null;
-        if (header) {
-            spanContext = SpanContext.fromString(this._decodedValue(header));
-            let baggage = {};
-            for (let key in carrier) {
-                if (carrier.hasOwnProperty(key) && Utils.startsWith(key, this._baggagePrefix)) {
+        // $FlowIgnore - I just want an empty span context.
+        let spanContext = new SpanContext();
+        let baggage = {};
+
+        for (let key in carrier) {
+            if (carrier.hasOwnProperty(key)) {
+                if (key === this._contextKey) {
```

Comment:
before comparing, convert the key to lower case, since HTTP does not guarantee that the case is preserved